tests/benchmark/test_anta.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""Benchmark tests for ANTA."""
+
+import asyncio
+import logging
+from unittest.mock import patch
+
+import pytest
+import respx
+from pytest_codspeed import BenchmarkFixture
+
+from anta.catalog import AntaCatalog
+from anta.inventory import AntaInventory
+from anta.result_manager import ResultManager
+from anta.result_manager.models import AntaTestStatus
+from anta.runner import main
+
+from .utils import collect, collect_commands
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "inventory",
+    [
+        pytest.param({"count": 1, "disable_cache": True, "reachable": False}, id="1 device"),
+        pytest.param({"count": 2, "disable_cache": True, "reachable": False}, id="2 devices"),
+    ],
+    indirect=True,
+)
+def test_anta_dry_run(benchmark: BenchmarkFixture, catalog: AntaCatalog, inventory: AntaInventory) -> None:
+    """Test and benchmark ANTA in Dry-Run Mode."""
+    # Disable logging during ANTA execution to avoid having these function time in benchmarks
+    logging.disable()
+
+    def bench() -> ResultManager:
+        """Need to wrap the ANTA Runner to instantiate a new ResultManger for each benchmark run."""
+        manager = ResultManager()
+        asyncio.run(main(manager, inventory, catalog, dry_run=True))
+        return manager
+
+    manager = benchmark(lambda: bench())  # pylint: disable=unnecessary-lambda
```

Comment:
Why can't we just do `manager = benchmark(bench)`?

Reply:
Good catch, I got lost when trying different syntaxes